src/control/lib/hardware/topology.go

```diff
@@ -68,6 +68,101 @@ func (t *Topology) NumCoresPerNUMA() int {
 	return 0
 }
 
+// AddDevice adds a device to the topology.
+func (t *Topology) AddDevice(numaID uint, device *PCIDevice) {
+	if t == nil || device == nil {
+		return
+	}
```

Comment:
Per our conversation in the other PR, I'm a little concerned that this pattern of ignoring programming errors may lead to confusing behavior or subtle bugs, particularly in light of the fact that this is all new code being used to explore new hardware topologies. I am sympathetic to the position that adding error handing on the part of the caller is less elegant, and I'd be open to the idea of using `panic()` in some of these cases if we're pretty sure that that these are "can't happen" situations.

Reply:
My thought was that the action is meaningless on a nil caller or nil input. In theory it shouldn't happen, but at this low a level it's always a little hard to guarantee caller behavior. Maybe it should just return an error, even if that looks ugly.

Reply:
I thought I'd save this for a follow-on, but I ended up taking your advice to solve a bug involving a corner case for the Topology merge.  Lesson learned.